build/xtask/src/dist.rs

```diff
@@ -2524,28 +2519,74 @@ fn write_elf(
     // Page align the start of sections data.
     let sections_data_offset = (program_headers_offset + 0xfff) & !0xfff;
 
+    // Page align the Section Header String Table.
+    let shstrtab_offset =
+        (sections_data_offset + sections_data.len() + 0xfff) & !0xfff;
+
+    // Create a Section Header String Table, to hold the actual section
+    // names.
+    let mut shstrtab = Vec::new();
+    shstrtab.push(0x00 as u8);
+    shstrtab.extend_from_slice(".text".as_bytes());
+    shstrtab.push(0x00 as u8);
+    shstrtab.extend_from_slice(".shstrtab".as_bytes());
+    shstrtab.push(0x00 as u8);
+    shstrtab.shrink_to_fit();
+
+    // Page align the Section Headers.
+    let sh_data_offset = (shstrtab_offset + shstrtab.len() + 0xfff) & !0xfff;
+
+    // We need to create a null section, as without this, objcopy fails to locate
+    // any section headers, and aborts.
+    let mut null_section_header = SectionHeader::new();
+    null_section_header.sh_type = SHT_NULL;
+
+    // We create a fake '.text' section to guide objcopy toward the instruction
+    // bitstream.
+    let mut text_section_header = SectionHeader::new();
+    text_section_header.sh_type = SHT_PROGBITS;
+    text_section_header.sh_flags = (SHF_ALLOC | SHF_EXECINSTR) as u64;
```

Comment:
Should this be marked as execute? The entire section is not really execute.

Reply:
At this point of packaging, we don't really have the ability to distinguish between sections to be loaded or not. My understanding of the ELF documentation is that the .text section mandates SHF_ALLOC & SHF_EXECINSTR: https://man7.org/linux/man-pages/man5/elf.5.html